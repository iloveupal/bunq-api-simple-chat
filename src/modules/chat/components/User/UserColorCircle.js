/**
 * This component helps us visually identify users because colors are easily distinguishable.
 * The idea is not new, you can see this stuff in Telegram, for example.
 *
 */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { stringToColor } from 'Utils/color';


const getInitials = (name) => {
    let initials = name.match(/\b\w/g) || [];

    return initials.join('');
};

const ColorCircle = styled.div`
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
    width: ${(props) => props.small ? '20px' : '40px'};
    height: ${(props) => props.small ? '20px' : '40px'};
    border-radius: ${(props) => props.small ? '10px' : '20px'};
    background-color: ${(props) => props.color};
    font-weight: bold;
    font-family: sans-serif;
    font-size: ${(props) => props.small ? '10px' : '14px'};
    color: white;
    cursor: pointer;
`;


export default class UserColorCircle extends PureComponent {
    static propTypes = {
        name: PropTypes.string,
        id: PropTypes.string.isRequired,
        small: PropTypes.bool,
    };

    render () {
        return (
            <ColorCircle
                color={stringToColor(this.props.id)}
                small={!!this.props.small}
                title={this.props.name}
            >
                {this.props.name && getInitials(this.props.name)}
            </ColorCircle>
        )
    }
}