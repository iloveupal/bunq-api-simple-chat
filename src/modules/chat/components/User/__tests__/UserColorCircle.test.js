import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import UserColorCircle from '../UserColorCircle';


describe('UserColorCircle', () => {
    it('should match snapshot', () => {
        const wrapper = mount(
            <UserColorCircle id={'1'} name={'Test'} />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('small. should match snapshot', () => {
        const wrapper = mount(
            <UserColorCircle id={'1'} name={'Test'} small/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});