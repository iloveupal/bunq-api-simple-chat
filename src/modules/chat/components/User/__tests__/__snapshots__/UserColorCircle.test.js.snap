// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserColorCircle should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 40px;
  height: 40px;
  border-radius: 20px;
  background-color: #eeeeee;
  font-weight: bold;
  font-family: sans-serif;
  font-size: 14px;
  color: white;
  cursor: pointer;
}

<UserColorCircle
  id="1"
  name="Test"
>
  <UserColorCircle__ColorCircle
    color="#eeeeee"
    small={false}
    title="Test"
  >
    <StyledComponent
      color="#eeeeee"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:flex;flex-flow:row nowrap;justify-content:center;align-items:center;width:",
              [Function],
              ";height:",
              [Function],
              ";border-radius:",
              [Function],
              ";background-color:",
              [Function],
              ";font-weight:bold;font-family:sans-serif;font-size:",
              [Function],
              ";color:white;cursor:pointer;",
            ],
          },
          "displayName": "UserColorCircle__ColorCircle",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      small={false}
      title="Test"
    >
      <div
        className="c0"
        color="#eeeeee"
        title="Test"
      >
        T
      </div>
    </StyledComponent>
  </UserColorCircle__ColorCircle>
</UserColorCircle>
`;

exports[`UserColorCircle small. should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
  height: 20px;
  border-radius: 10px;
  background-color: #eeeeee;
  font-weight: bold;
  font-family: sans-serif;
  font-size: 10px;
  color: white;
  cursor: pointer;
}

<UserColorCircle
  id="1"
  name="Test"
  small={true}
>
  <UserColorCircle__ColorCircle
    color="#eeeeee"
    small={true}
    title="Test"
  >
    <StyledComponent
      color="#eeeeee"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "display:flex;flex-flow:row nowrap;justify-content:center;align-items:center;width:",
              [Function],
              ";height:",
              [Function],
              ";border-radius:",
              [Function],
              ";background-color:",
              [Function],
              ";font-weight:bold;font-family:sans-serif;font-size:",
              [Function],
              ";color:white;cursor:pointer;",
            ],
          },
          "displayName": "UserColorCircle__ColorCircle",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      small={true}
      title="Test"
    >
      <div
        className="c0"
        color="#eeeeee"
        title="Test"
      >
        T
      </div>
    </StyledComponent>
  </UserColorCircle__ColorCircle>
</UserColorCircle>
`;
